src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -32,8 +32,8 @@ public class Program
         private const string DotNetSetupRegistryKey = @"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\";
         private const int Net462ReleasedVersion = 394802;
 
-
-        private static readonly string ThisExecutableName = typeof(Program).Assembly.GetName().Name;
+        internal static readonly Assembly NuGetExeAssembly = typeof(Program).Assembly;
```

Comment:
Given that these changes are not going to make any localization work, should we just delayed that for the future PR?

It seems like the new branch you have mostly undoes these, https://github.com/NuGet/NuGet.Client/compare/dev-dominofire-nugetexe-defaultloc...dev-dominofire-nugetexe-loc-mixed-embedding

Reply:
I moved all required changes into this PR and left remaining changes for NuGet.Commands in the other PR.